include/class_loader/class_loader_core.hpp

```diff
@@ -41,12 +41,16 @@
 #include <utility>
 #include <vector>
 
-#include "Poco/SharedLibrary.h"
-
 #include "class_loader/exceptions.hpp"
 #include "class_loader/meta_object.hpp"
 #include "class_loader/visibility_control.hpp"
 
+// forward declaration
+namespace Poco
+{
+  class SharedLibrary;
+}
```

Comment:
```suggestion
}  // namespace Poco
```

It's a nitpick, but I believe cpplint would complain here, though I don't know if we run class_loader through that linter off-hand.